RFCXXX-Native-Json-Adapters.md

```diff
@@ -0,0 +1,275 @@
+---
+RFC: RFCXXX-Native-Json-Adapters.md
+Author: Jim Truher
+Status: Draft
+SupercededBy: 
+Version: 0.9
+Area: engine
+Comments Due: May 1, 2023
+Plan to implement: Yes
+---
+
+# Improving the User Experience of Native Applications by suggesting improved pipelines
+
+## Motivation
+
+As a user I can interact with native applications and have a more PowerShell-like experience.
+
+### Native Applications which Emit Structured Data
+
+Native command execution usually results in textual output which is a fairly poor experience for our users.
+Some of our users use tools like `Crescendo` to enable those tools to "play nice" in the PowerShell arena,
+but the problem of output handling is still troublesome.
+
+Some applications (docker, kubectl, others) can emit `JSON` which is much easier for our users to convert to useful results in PowerShell,
+but the user still has to know the incantation to have the tool emit `JSON` and then they must pipe that output to `Convert-FromJson`.
+We can improve the experience for the user if we can help them discover these tools and additional parameters.
+If the user types:
+
+```powershell
+PS> docker image ls
+```
+
+we can suggest:
```

Comment:
Need a way to not make the same suggestion multiple times if the user doesn't accept it so it doesn't become noisy.  Not sure yet if this should be per feedback provider or part of the feedback subsystem.

Reply:
if the accepted suggestion gets in history that can be a way to not repeat the suggestion again and rely on predictors to surface/remind that behavior